apps/system/test/unit/app_window_test.js

```diff
@@ -3077,6 +3106,27 @@ suite('system/AppWindow', function() {
     });
   });
 
+  test('manifestchange', function() {
+    var app1 = new AppWindow(fakeWrapperConfig);
+
+    app1.handleEvent({
+      type: 'mozbrowsermanifestchange',
+      detail: {
+        href: 'http://example.com/manifest.json'
+      }
+    });
+
+    assert.equal(app1.webManifestURL, 'http://example.com/manifest.json');
+  });
+
+  test('_fetchManifest()', function(done) {
```

Comment:
This is too internal. It's a bit hacky but I'd rather go with something like
```
    var app1 = new AppWindow(fakeWrapperConfig);
    app1.handleEvent({
      type: 'mozbrowsermanifestchange',
      detail: {
        href: 'http://example.com/manifest.json'
      }
    });

    Promise.resolve().then(function() {
      assert.equal(app1.name, 'App');
      done();
    });

```
You can probably merge the 2 tests this way too.